include/deal.II/grid/grid_in.h

```diff
@@ -663,8 +663,25 @@ class GridIn
 
 
   DeclException0(ExcGmshNoCellInformation);
+  
+  /**
+   * 
+   */
+  std::map< std::string, std::map<int,double> > get_field_data()
+  {
+      
+      return this->field_data;
+  }
```

Comment:
Is there a good reason that this function returns a non-`const` reference object? If it returned a `const` reference, you could also make the function itself `const`.

Reply:
This is still an open comment. What you're doing here with this function is return a *copy* of the object. That's an expensive operation. An easier approach would be to return a `const` reference (i.e., in essence a pointer to the object you're storing).

Reply:
This is still an open comment.